massa-execution-worker/src/execution.rs

```diff
@@ -248,7 +248,56 @@ impl ExecutionState {
         if slot > max_slot {
             panic!("cannot execute at a slot beyond active cursor + 1");
         }
+    }
+
+    /// Computes the index of a given slot in the active history
+    fn get_active_index(&self, slot: Slot) -> Option<usize> {
+        let current = self.active_cursor.period * (self.config.thread_count as u64)
+            + (self.active_cursor.thread as u64);
+        let asked = slot.period * (self.config.thread_count as u64) + (slot.thread as u64);
+        current.checked_sub(asked).and_then(|v| v.try_into().ok())
+    }
+
+    /// Lazily query (from end to beginning) the active balance of an address at a given slot.
+    /// Returns None if the address balance could not be determined from the active history.
+    ///
+    /// NOTE: temporary, needs to be done in the speculative ledger
+    pub fn fetch_active_history_balance(
+        &self,
+        slot: Slot,
+        addr: &Address,
+    ) -> HistorySearchResult<Amount> {
+        self.verify_active_slot(slot);
+
+        if let Some(n) = self.get_active_index(slot) {
+            let iter = self.active_history.iter().rev().skip(n);
```

Comment:
I think it's

```
let iter = self.active_history.iter().skip(n).rev();
```

Reply:
Are you sure? `active_history` has the newest element A at the back, so if we want to fetch the data of a given slot B the `skip` should happen after the `rev` because we are skipping from A to B, and parcouring the slots that come before B to determine the value. Or am I getting it wrong?

Reply:
FI, this depends on which implementation of `get_active_index` we are using: https://github.com/massalabs/massa/pull/2590#discussion_r883769585

Reply:
Done.